### PR TITLE
RTCDataChannelStats

### PR DESCRIPTION
**Pull request by zhuker**
*Description of changes:*

Placeholder PR for RTCDataChannelStats  https://www.w3.org/TR/webrtc-stats/#dom-rtcdatachannelstats

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/include/com/amazonaws/kinesis/video/webrtcclient/Stats.h [hunk 1]
```diff
@@ -489,6 +489,31 @@ typedef struct {
...
+
+/**
+ * Reference: https://www.w3.org/TR/webrtc-stats/#dom-rtcdatachannelstats
+ */
+struct {
+    DOMString label;                //!< The "label" value of the RTCDataChannel object.
+    DOMString protocol;             //!< The "protocol" value of the RTCDataChannel object.
+    long dataChannelIdentifier;     //!< The "id" attribute of the RTCDataChannel object.
+    DOMString transportId;          //!< A stats object reference for the transport used to carry this datachannel.
+    enum RTCDataChannelState state; //!< The "readyState" value of the RTCDataChannel object.
```
Comment (disa6302) [inline src/include/com/amazonaws/kinesis/video/webrtcclient/Stats.h:510]: Can we typedef the enum and also have the enum type in all caps? 
`RTC_DATA_CHANNEL_STATE`

File: src/include/com/amazonaws/kinesis/video/webrtcclient/Stats.h [hunk 2]
```diff
@@ -489,6 +489,31 @@ typedef struct {
...
+    RTC_DATA_CHANNEL_STATE_CLOSED
+};
+
+/**
+ * Reference: https://www.w3.org/TR/webrtc-stats/#dom-rtcdatachannelstats
+ */
+struct {
+    DOMString label;                //!< The "label" value of the RTCDataChannel object.
+    DOMString protocol;             //!< The "protocol" value of the RTCDataChannel object.
+    long dataChannelIdentifier;     //!< The "id" attribute of the RTCDataChannel object.
```
Comment (disa6302) [inline src/include/com/amazonaws/kinesis/video/webrtcclient/Stats.h:508]: Long is INT32 in our stack

File: src/include/com/amazonaws/kinesis/video/webrtcclient/Stats.h (hunk 1 above)
Comment (zhuker) [reply src/include/com/amazonaws/kinesis/video/webrtcclient/Stats.h:510]: yes

File: src/include/com/amazonaws/kinesis/video/webrtcclient/Stats.h [hunk 3]
```diff
@@ -489,6 +489,31 @@ typedef struct {
...
+    RTC_DATA_CHANNEL_STATE_CONNECTING,
+    RTC_DATA_CHANNEL_STATE_OPEN,
+    RTC_DATA_CHANNEL_STATE_CLOSING,
+    RTC_DATA_CHANNEL_STATE_CLOSED
+} RTC_DATA_CHANNEL_STATE;
+
+/**
+ * Reference: https://www.w3.org/TR/webrtc-stats/#dom-rtcdatachannelstats
+ */
+struct {
```
Comment (disa6302) [inline src/include/com/amazonaws/kinesis/video/webrtcclient/Stats.h:505]: Sorry. Missed this earlier. But, this needs to typedef'd as well. That is probably why the CI is failing with link errors.

Comment (zhuker) [reply src/include/com/amazonaws/kinesis/video/webrtcclient/Stats.h:505]: its embarrassing :-/

**Comment by zhuker** [comment]
@disa6302 very weird CI failure can you take a look?

**Comment by disa6302** [comment]
> @disa6302 very weird CI failure can you take a look?

@zhuker , the failure seems weird. I have restarted the build. Let us see if it passes.

**Comment by disa6302** [comment]
@zhuker , 

Maybe something funky happened then. But the build passed now :D
Is there anything you would like to add, or should I go ahead and merge this? 

Once this is merged, I can rebase my branch with stats change and push a PR.

**Comment by disa6302** [comment]
Oh, and also, can you squash the commits?

**Comment by zhuker** [comment]
@disa6302 my plan was to work on this but if you already have it then that's all it is

**Comment by zhuker** [comment]
squashed
merge it
@disa6302

**Comment by disa6302** [comment]
> @disa6302 my plan was to work on this but if you already have it then that's all it is

Cool then! I can merge this in once the CI passed and create a PR with my changes soon. :)

Thanks for your contribution @zhuker !

**Pull request merged**